plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/model/VFSFile.java

```diff
@@ -186,4 +188,12 @@ public int hashCode() {
   public EntityType getEntityType(){
     return EntityType.VFS_FILE;
   }
+
+  @Override
+  public String getNameDecoded() throws FileSystemException {
+    if ( StringUtils.isNotEmpty( getName() ) ) {
+      return UriParser.decode( getName() );
+    }
+    return getName();
```

Comment:
Catch ApacheVFS `FileSystemException` and wrap it in a `org.pentaho.di.plugins.fileopensave.api.providers.exception.FileException`.